lib/OpenQA/Task/Asset/Limit.pm

```diff
@@ -86,13 +86,17 @@ sub _limit {
 
         my $age_in_seconds = ($now->epoch() - DateTime::Format::Pg->parse_datetime($asset->{t_created})->epoch());
         my $asset_name     = $asset->{name};
-        if ($age_in_seconds >= $untracked_assets_storage_duration || !$asset->{size}) {
+
+        if ($age_in_seconds >= $untracked_assets_storage_duration) {
             my $age_in_days   = $age_in_seconds / $seconds_per_day;
             my $limit_in_days = $untracked_assets_storage_duration / $seconds_per_day;
             _remove_if($schema, $asset,
 "Removing asset $asset_name (not in any group, age ($age_in_days days) exceeds limit ($limit_in_days days)"
             );
         }
+        elsif (!$asset->{size} && $age_in_seconds >= $seconds_per_day) {
```

Comment:
I'd say just drop this case, which is probably what Marius was referring to. This is just adding complexity. It can be removed as usual regardless of size.